cocotb/triggers.py

```diff
@@ -161,11 +161,37 @@ def unprime(self):
 class Timer(GPITrigger):
     """Fires after the specified simulation time period has elapsed."""
     def __init__(self, time_ps, units=None):
+        """
+        Args:
+           time_ps: The time value. This can also be a 
+               :class:`fractions.Fraction` or a :class:`decimal.Decimal` instance.
+               Note that despite the name this is not actually in picoseconds
+               but depends on the *units* argument.
+           units (str or None, optional): One of
+               ``None``, ``'fs'``, ``'ps'``, ``'ns'``, ``'us'``, ``'ms'``, ``'sec'``.
+               When no *units* is given (``None``) the timestep is determined by
+               the simulator.
+
+               See Also:
+                   :func:`~cocotb.utils.get_sim_steps`
+
+        Example:
+
+        .. code-block:: python
+    
+            yield Timer(100, units='us')
+            # alternate spelling 1:
+            from fractions import Fraction
+            yield Timer(Fraction(1, 10000), units='sec')
+            # alternate spelling 2:
+            from decimal import Decimal
+            yield Timer(Decimal('1e-9'), units='sec')
```

Comment:
Examples:

            >>> yield Timer(100, units='ps')

            The time can also be a ``float``:

            >>> yield Timer(100e-9, units='sec')

            which is particularly convenient when working with frequencies:

            >>> freq = 10e6  # 10 MHz
            >>> yield Timer(1 / freq, units='sec')

        	Other builtin exact numeric types can be used too:

            >>> from fractions import Fraction
            >>> yield Timer(Fraction(1, 10), units='ns')

            >>> from decimal import Decimal
            >>> yield Timer(Decimal('100e-9'), units='sec')

            These are most useful when using computed durations while
            avoiding floating point inaccuracies.

Reply:
I put this in.